core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -116,41 +185,167 @@ public SimpleQueryStringBuilder flags(SimpleQueryStringFlag... flags) {
         return this;
     }
 
-    public SimpleQueryStringBuilder analyzeWildcard(boolean analyzeWildcard) {
-        this.analyzeWildcard = analyzeWildcard;
+    /** Returns whether query parsing should be lenient. */
+    public Boolean lenient() {
+        return this.settings.lenient();
+    }
+
+    /** Specifies whether wildcards should be analyzed. Defaults to false. */
+    public SimpleQueryStringBuilder analyzeWildcard(Boolean analyzeWildcard) {
+        this.settings.analyzeWildcard(analyzeWildcard);
         return this;
     }
 
+    /** Returns whether wildcards should by analyzed. */
+    public Boolean analyzeWildcard() {
+        return this.settings.analyzeWildcard();
+    }
+
+    /**
+     * Specifies the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
     public SimpleQueryStringBuilder minimumShouldMatch(String minimumShouldMatch) {
         this.minimumShouldMatch = minimumShouldMatch;
         return this;
     }
 
+    /**
+     * Returns the minimumShouldMatch to apply to the resulting query should
+     * that be a Boolean query.
+     */
+    public String minimumShouldMatch() {
+        return minimumShouldMatch;
+    }
+
+    /** Returns an exception if any mandatory fields are null. */
+    @Override
+    public QueryValidationException validate() {
+        QueryValidationException validationException = null;
+
+        // Query text is required
+        if (queryText == null) {
+            validationException = QueryValidationException.addValidationError("[" + SimpleQueryStringBuilder.NAME + "] query text missing",
+                    validationException);
+        }
+
+        return validationException;
+    }
+
+    /** Returns the lucene query parsed from the parameters. */
+    @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        // Use the default field (_all) if no fields specified
+        if (fieldsAndWeights.isEmpty()) {
+            String field = parseContext.defaultField();
+            fieldsAndWeights.put(field, 1.0F);
+        }
+
+        // Use standard analyzer by default if none specified
+        Analyzer luceneAnalyzer;
+        if (analyzer == null) {
+            luceneAnalyzer = parseContext.mapperService().searchAnalyzer();
+        } else {
+            luceneAnalyzer = parseContext.analysisService().analyzer(analyzer);
+            if (luceneAnalyzer == null) {
+                throw new QueryParsingException(parseContext, "[" + SimpleQueryStringBuilder.NAME + "] analyzer [" + analyzer
+                        + "] not found");
+            }
+
+        }
+        settings.addDefaults();
+        SimpleQueryParser sqp = new SimpleQueryParser(luceneAnalyzer, fieldsAndWeights, flags, settings);
+
+        if (defaultOperator != null) {
+            switch (defaultOperator) {
+            case OR:
+                sqp.setDefaultOperator(Occur.SHOULD);
+                break;
+            case AND:
```

Comment:
Whatever is more readable. Didn't check the rest of the code base before writing this switch.